### PR TITLE
LPS-202061 Periodical configuration to guest user add object entries

### PR DESCRIPTION
**Pull request by thallescezar**
Related [ticket](https://liferay.atlassian.net/browse/LPS-202061)

### CODE REVIEW

**Comment by thallescezar** [comment]
ci:test:sf

**Comment by thallescezar** [comment]
ci:test:sf

**Comment by thallescezar** [comment]
ci:test:stable

**Comment by thallescezar** [comment]
ci:test:object

**Comment by thallescezar** [comment]
ci:test:relevant

**Comment by thallescezar** [comment]
ci:test:sf

**Comment by thallescezar** [comment]
ci:test:stable

**Comment by thallescezar** [comment]
ci:test:relevant

**Comment by thallescezar** [comment]
ci:test:object

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryServiceTest.java [hunk 1]
```diff
@@ -640,6 +655,96 @@ public void testSearchObjectEntries() throws Exception {
...
 
+	@Test
+	public void testValidateMaximumNumberOfObjectEntries() throws Exception {
+		_setUser(_guestUser);
+
+		HashMapDictionaryBuilder.HashMapDictionaryWrapper<String, Object>
+			objectConfigurationDictionary = HashMapDictionaryBuilder.create(
+				Collections.emptyMap());
+
+		HashMapDictionaryBuilder.put(
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryServiceTest.java:666]: Hey @thallescezar , can you put these values in the initialization of HashMapDictionaryBuilder? You can follow this [pattern](https://github.com/liferay/liferay-portal/blob/04aa57bf2c79629b8f3bff76e9102e882a8e01f7/modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/deployer/ObjectDefinitionDeployerImpl.java#L289-L306), this way we avoid these many lines invoking put method.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryServiceTest.java [hunk 2]
```diff
@@ -640,6 +655,96 @@ public void testSearchObjectEntries() throws Exception {
...
+		ConfigurationTestUtil.saveConfiguration(
+			ObjectConfiguration.class.getName(),
+			objectConfigurationDictionary.build());
+
+		LocalDate beginningDate = LocalDate.now(
+		).minusDays(
+			7
+		);
+
+		Date createdDate = Date.from(
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryServiceTest.java:720]: Please rename this variable name to `createDate`

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryServiceTest.java [hunk 3]
```diff
@@ -640,6 +655,96 @@ public void testSearchObjectEntries() throws Exception {
...
+
+		HashMapDictionaryBuilder.put(
+			"maximumNumberOfGuestUserObjectEntriesPerObjectDefinition", 2);
+		HashMapDictionaryBuilder.put("timeScale", "weeks");
+
+		ConfigurationTestUtil.saveConfiguration(
+			ObjectConfiguration.class.getName(),
+			objectConfigurationDictionary.build());
+
+		LocalDate beginningDate = LocalDate.now(
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryServiceTest.java:715]: I think you can remove the minusDays(7) from this line and rename the variable to localDate, then you apply the minusDays(7) in the next line. Is this possible?

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryServiceTest.java [hunk 4]
```diff
@@ -698,6 +801,9 @@ private void _testDeleteObjectEntry(User ownerUser, User user)
 	@Inject
 	private ObjectEntryLocalService _objectEntryLocalService;
 
+	@Inject
+	private ObjectEntryPersistence _objectEntryPersistence;
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryServiceTest.java:805]: Please remove this reference and just use `_objectEntryLocalService`

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryServiceTest.java [hunk 5]
```diff
@@ -640,6 +656,93 @@ public void testSearchObjectEntries() throws Exception {
...
+
+		ObjectEntry objectEntry = _objectEntryService.addObjectEntry(
+			0, _objectDefinition.getObjectDefinitionId(),
+			Collections.emptyMap(),
+			ServiceContextTestUtil.getServiceContext(
+				TestPropsValues.getGroupId(), _guestUser.getUserId()));
+
+		objectEntry.setCreateDate(createDate);
+
+		_objectEntryPersistence.update(objectEntry);
```
[the remark this answers is not in the document]
Comment (thallescezar) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryServiceTest.java:731]: I have tried use this method before and it does not work on this case, then I talked with Feliphe and we decided to use the Persistence.

**Comment by guilhermedcamacho** [comment]
ci:test:sf

**Comment by thallescezar** [comment]
ci:test:sf

**Comment by thallescezar** [comment]
ci:test:sf

**Comment by thallescezar** [comment]
ci:test:sf

**Comment by guilhermedcamacho** [comment]
ci:test:relevant

**Review by guilhermedcamacho — APPROVED** [review]

**Comment by guilhermedcamacho** [comment]
ci:test:relevant

**Comment by guilhermedcamacho** [comment]
ci:test:object

**Comment by guilhermedcamacho** [comment]
Hey @thallescezar , there are some [tests](https://testray.liferay.com/home/-/testray/case_results/2474285321/history?tab=result&hideFilterPin=true) failing after your changes, can you check them please?

**Comment by guilhermedcamacho** [comment]
ci:stop

**Comment by guilhermedcamacho** [comment]
ci:stop

**Comment by thallescezar** [comment]
ci:test:sf

**Comment by thallescezar** [comment]
ci:test:sf

**Comment by thallescezar** [comment]
ci:test:object

**Comment by thallescezar** [comment]
ci:test:relevant

**Comment by thallescezar** [comment]
ci:test:stable

**Comment by guilhermedcamacho** [comment]
Manually forwarded to [brianchandotcom#145286](https://github.com/brianchandotcom/liferay-portal/pull/145286)

**Pull request closed without merging**